sympy/integrals/transforms.py

```diff
@@ -1825,19 +1744,74 @@ def _collapse_extra(self, extra):
                 'Laplace', None, 'No combined convergence.')
         return plane, cond
 
-    def _try_directly(self, **hints):
-        fn = self.function
-        debug('----> _try_directly: %s'%(fn, ))
+    def doit(self, **hints):
+        """
+        Try to evaluate the transform in closed form.
+
+        Explanation
+        ===========
+
+        Standard hints are the following:
+        - ``noconds``:  if True, do not return convergence conditions. This is
+        the default behaviour.
+        """
+        _noconds = hints.get('noconds', False)
+        _simplify = hints.get('simplify', True)
+
+        debug('[LT doit] (%s, %s, %s)'%(self.function,
+                                        self.function_variable,
+                                        self.transform_variable))
+
         t_ = self.function_variable
         s_ = self.transform_variable
-        LT = None
-        if not fn.is_Add:
-            fn = expand_mul(fn)
+        fn = self.function
+
+        terms = Add.make_args(fn)
+        results = []
+        for ff in terms:
+            k, f = ff.as_independent(t_, as_Add=False)
+            ft = (f, S.NegativeInfinity, True)
+            d, r = _laplace_apply_simple_rules(ft, t_, s_)
+            if d:
+                results.append((k*r[0], r[1], r[2]))
+                continue
+            d, r = _laplace_apply_prog_rules(ft, t_, s_)
+            if d:
+                results.append((k*r[0], r[1], r[2]))
+                continue
+            d, r = _laplace_expand(ft, t_, s_)
+            if d:
+                results.append((k*r[0], r[1], r[2]))
+                continue
+            T = None
             try:
-                LT = self._compute_transform(fn, t_, s_, **hints)
+                T = self._compute_transform(f, t_, s_, **hints)
             except IntegralTransformError:
-                LT = None
-        return fn, LT
+                T = None
+            if T is not None:
+                results.append((k*T[0], T[1], T[2]))
+            else:
+                results.append((k*LaplaceTransform(f, t_, s_),
+                                S.NegativeInfinity, True))
+
+        r = []
+        p = []
+        c = []
+        for res in results:
+            r.append(res[0])
+            p.append(res[1])
+            c.append(res[2])
```

Comment:
Looking at this more closely the signature of these rule functions doesn't really make any sense. They all take a tuple `ft = (f, p, c)` as input argument which they immediately unpack. The values of `p` and `c` are always the same on input and although the rule functions use `Max` and `And` this is pointless because e.g. `Max(-oo, pc)` will always just give `pc`. The other place the `p` and `c` arguments are used is when returning `False, (f, p, c)` but there's no point in returning a value along with the False because the value will always be ignored.

I thought that there might be some place where the input plane or convergence conditions would be something other than `-oo, True` but clearly as shown above there is not. There are places that combine convergence conditions with `Max/And` but it happens inside these functions when they call `doit` recursively and so these input values are redundant.

Instead these functions should just take `f` as input and then always return a tuple `(fs, p, c)` or `None` if the rule does not apply. Then this function is just like:
```python
result = _laplace_apply_simple_rules(f, t_, s_)
if result is not None:
    ...
```
Actually this code above is a good candidate for the Walrus operator:
```python
        terms = Add.make_args(fn)
        terms_s = []
        planes = []
        conditions = []

        for term in terms:

            coefficient, ft = term.as_independent(t_, as_Add=False)

            if result := _laplace_apply_simple_rules(ft, t_, s_) is not None):
                pass
            elif result := _laplace_apply_prog_rules(ft, t_, s_) is not None):
                pass
            elif result := _laplace_expand(ft, t_, s_) is not None):
                pass
            elif result := _laplace_integrate(f, t_, s_, **hints) is not None:
                pass
            else:
                f_lt = LaplaceTransform(f, t_, s_)
                result = (f_lt, S.NegativeInfinity, S.true)

            fs, p, c = result
            terms_s.append(coefficient * fs)
            planes.append(p)
            conditions.append(c)
```

Reply:
You are right; originally I had plans to be able to use non-ideal planes and conditions as input, but that is, as you say, not necessary anymore. I'll let the Walrus loose ...